play-services-core/src/main/kotlin/com/google/android/gms/locationsharingreporter/service/LocationSharingReporterApiService.kt

```diff
@@ -0,0 +1,240 @@
+        Feature("START_LOCATION_REPORTING", 1),
+        Feature("STOP_LOCATION_REPORTING", 1),
+        Feature("GET_REPORTING_ISSUES", 1),
+        Feature("UPDATE_NOTICE_STATE", 1),
+        Feature("LOCATION_SHARING", 1),
+)
+
+class LocationSharingReporterApiService : BaseService(TAG, GmsService.LOCATION_SHARING_REPORTER) {
+
+    override fun handleServiceRequest(callback: IGmsCallbacks, request: GetServiceRequest, service: GmsService?) {
+        Log.d(TAG, "handleServiceRequest: $request")
+        val packageName = PackageUtils.getAndCheckCallingPackage(this, request.packageName)
+                ?: throw IllegalArgumentException("Missing package name")
+        val callingPackageName = PackageUtils.getCallingPackage(this) ?: packageName
+        callback.onPostInitCompleteWithConnectionInfo(
+                CommonStatusCodes.SUCCESS,
+                LocationSharingReporterApiServiceImpl(this, callingPackageName, lifecycle).asBinder(),
+                ConnectionInfo().apply { features = FEATURES }
+        )
+    }
+
+    override fun onDestroy() {
+        Log.d(TAG, "onDestroy: ")
+        super.onDestroy()
+    }
+}
+
+class LocationSharingReporterApiServiceImpl(
+        val context: Context,
+        val callingPackageName: String,
+        override val lifecycle: Lifecycle
+) : ILocationSharingReporterService.Stub(), LifecycleOwner {
+
+    override fun updateLocation(callback: ILocationUploadCallbacks, account: Account, request: LocationUploadRequest, apiMetadata: ApiMetadata) {
+        Log.d(TAG, "Not yet implemented: updateLocation called with account: ${account.name}")
+    }
+
+    override fun getReportingStatus(callback: ILocationReportingStatusCallbacks, account: Account, apiMetadata: ApiMetadata) {
+        Log.d(TAG, "Not yet implemented: getReportingStatus called with account: ${account.name}")
+    }
+
+    override fun syncReportingStatus(callback: IStatusCallback, account: Account, apiMetadata: ApiMetadata) {
+        Log.d(TAG, "Not yet implemented: syncReportingStatus called with account: ${account.name}")
+    }
+
+    override fun periodicLocationUpload(callback: IStatusCallback, account: Account, request: PeriodicLocationUploadRequest, apiMetadata: ApiMetadata) {
+        Log.d(TAG, "Not yet implemented: periodicLocationUpload called with account: ${account.name}, request: $request")
+    }
+
+    override fun startLocationReporting(callback: IStatusCallback, account: Account, request: StartLocationReportingRequest, apiMetadata: ApiMetadata) {
+        Log.d(TAG, "startLocationReporting called with account: ${account.name}, request: $request")
+        try {
+            validateGoogleAccount(account)
+            validateMakePrimaryOption(request.makePrimary)
+            validateReportingType(request.reportingType)
+
+            //Start location sharing, location sharing is enabled by default
+            sendLocationSharingEnable(true, account, context)
```

Comment:
This enables location sharing without user consent. Make sure there is user consent before enabling location sharing!